Revisit nginx docs (#787)
giantswarm/docs: src/content/guides/multi-nginx/index.md

Some use cases for this might be:

- An Ingress Controller that is behind an internal ELB for traffic between services within the VPC (or a group of peered VPCs)
- An Ingress Controller behind an ELB that already terminates SSL
- An Ingress Controller with different functionality or performance

__Note__ that if you are running multiple Ingress Controllers you need to annotate each Ingress with the appropriate class, e.g.

```yaml
kubernetes.io/ingress.class: "nginx"
```

or

```yaml
kubernetes.io/ingress.class: "nginx-internal"
```

Not specifying the annotation will lead to multiple ingress controllers claiming the same ingress. Specifying a value which does not match the class of any existing ingress controllers will result in all ingress controllers ignoring the ingress.

> Further note that if you are running additional Ingress Controllers you might need to configure them so their Ingress Class does not collide with the class of our default NGINX Ingress Controller. For the community supported NGINX Ingress Controller this is described in the [official documentation](https://kubernetes.github.io/ingress-nginx/user-guide/multiple-ingress/).

Reviewer: We no longer have default (preinstalled) nginx in most clusters https://docs.giantswarm.io/general/releases/#apps - how does that change this?

Maybe sth like this?
```suggestion
Additionally, please ensure the Ingress Class of each of your Ingress Controllers do not collide with each other. For the community supported NGINX Ingress Controller this is described in the [official documentation](https://kubernetes.github.io/ingress-nginx/user-guide/multiple-ingress/).
```

Most NGINX configuration options have NGINX-wide defaults. They can also be overriden on a per-Ingress resource level.

In each case below, one installs a second NGINX with a different NGINX global-only configuration. Ingress resources managed by this NGINX installation cannot be customized on a per-Ingress resource level.

> Further information on configuring NGINX Ingress Controller can be found in the [Services of type LoadBalancer]

Reviewer: ```suggestion
Further information on configuring NGINX Ingress Controller can be found in the article, [Services of type LoadBalancer]({{< relref "/content/guides/services-of-type-loadbalancer/index.md" >}}).
```
